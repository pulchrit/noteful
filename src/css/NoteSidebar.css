.nav {
    width: 30%;
    margin: 10px;
}

.go-back-button {
    /* display: block; */
    padding: 20px;
    width: 100%;
    margin-bottom: 5px;
    border: 1px solid black;
    border-radius: 2px;
}

.folder-name {
    font-size: 1.5em;
    text-align: center;
}