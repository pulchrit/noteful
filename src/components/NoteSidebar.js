import React from 'react';
import NotefulContext from './NotefulContext';
import "../css/NoteSidebar.css"

export default class NoteSidebar extends React.Component {

    static contextType = NotefulContext;
    
    render() {

        const getFolderName = () => {
            const thisNote = this.context.notes.find(note => note.id === this.props.match.params.noteId);
            const thisFolder = this.context.folders.find(folder => folder.id === thisNote.folderId);
            return thisFolder.name;
        }

        const handleClickGoBack = () => {
            this.props.history.goBack();
        }


        return (
            <nav className="nav">
                <button type='button' className="go-back-button" onClick={handleClickGoBack}>
                    Go back
                </button>

                <h2 className="folder-name">{getFolderName()}</h2>
            </nav>
        )
    }
}