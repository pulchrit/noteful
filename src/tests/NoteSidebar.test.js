import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import NoteSidebar from '../components/NoteSidebar';

const folderName = 'super';
const routeProps = 
    <BrowserRouter>
        <NoteSidebar folderName={folderName} onClickGoBack={() => routeProps.history.goBack()}/>
    </BrowserRouter>;
const onClickGoBack = () => routeProps.history.goBack();

describe("NoteSidebar component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <NoteSidebar />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI of the sidebar correctly', () => {
    const wrapper = shallow(<NoteSidebar folderName={folderName} onClickGoBack={onClickGoBack} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

 /*  it('renders UI correctly after click of "go back" button', () => {
    const wrapper = shallow(<NoteSidebar folderName={folderName} onClickGoBack={() => routeProps.history.goBack()} />);
    wrapper.find('.go-back-button').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  }); */
});

