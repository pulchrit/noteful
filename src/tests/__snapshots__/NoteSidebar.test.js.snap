// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteSidebar component renders UI of the sidebar correctly 1`] = `
<nav
  className="nav"
>
  <button
    className="go-back-button"
    onClick={[Function]}
    type="button"
  >
    Go back
  </button>
  <h2
    className="folder-name"
  >
    super
  </h2>
</nav>
`;
